### DISCUSSION TITLE
Abaqus input file in MOOSE

### DISCUSSION THREAD

**Discussion by sidharthsarmah** [idaholab/moose#21812] [Q&A General] [answered]
Hi everyone,I am very new to the MOOSE software and I want to extract the mesh obtained in ABAQUS to input it as a MOOSE generated file input to create the mesh and elements? or can you tell me how to use the ABAQUS model with mesh elements in moose, thanks!

**Comment by dschwen** [comment, marked as answer]
You can use `file = YourAbaqus.inp` in the `[Mesh]` block to load the geometry into MOOSE. There is no way (yet) of generating the entire input file from the (i.e. loading conditions, constitutive model parameters). You'll have to manually build the input. In many cases this is straight forward, especially since we offer support for Abaqus UMAT plugins.

**Reply by libinbabu1991** [reply]
Hello, 
The below support page says .inp mesh is accepted in MOOSE.
https://mooseframework.inl.gov/source/mesh/FileMesh.html

I created .inp mesh from ANSYS SpaceClaim and loaded into the [Mesh] block as advised on the above comment.
However I am getting the same error:

(moose) root@131-ENG-DESKTOP:~/projects/moose/examples/ex01_inputfile#  ./ex01-opt -i ex01.i


*** ERROR ***
Invalid dimension dim = 4

Can I get some support in this matter, please?


Thanks.

**Reply by GiudGiud** [reply]
Do you have a link to the inp file specifications? We need to know what 4 means for the mesh dimension.

**Reply by libinbabu1991** [reply]
[mesh2.txt](https://github.com/user-attachments/files/17883033/mesh2.txt)

Hello @GiudGiud , 
I couldn't find a proper documentation on .inp file. I copied the contents of a sample mesh file in .inp format to a text file and attached in this message. Please let me know if that helps.

Thanks

**Reply by sidharthsarmah** [reply]
The .inp should contain node represent by (x,y,z) coordinates, eg 1.  0.01.  0.001 0.0001 and the element generated by these nodes. How come the space coordinates has 4 dimensions?

**Reply by libinbabu1991** [reply]
Hello @sidharthsarmah , 
I am not sure. Is it possible to convert the .inp format to .e ?

**Reply by GiudGiud** [reply]
You can try meshio it s a Python library that might be able to convert it

**Reply by libinbabu1991** [reply]
Will Try. Thanks

**Comment by sidharthsarmah** [comment]
You can either check the example of an .e file and check similarly to convert your .inp into it.
But .inp file works for me. Just get the .inp done correctly.

**Reply by libinbabu1991** [reply]
I will check. Thanks

**Reply by hxz126** [reply]
It seems the node and element sets cannot be loaded into MOOSE correctly if there are duplicate nodes or elements in those sets via either the .e file obtained from meshio or the input file using type = FileMeshGenerator.

**Reply by GiudGiud** [reply]
>  if there are duplicate nodes or elements in those sets

That's very possible. Do you have a MWE and we can document the issue?

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , I have an issue in running making file in clusters:
```
Creating Unity /home/sidharth/projects/def-jsong/sidharth/moose/modules/solid_mechanics/build/unity_src/materials_Unity.C
Compiling C++ (in opt mode) /home/sidharth/projects/def-jsong/sidharth/moose/modules/solid_mechanics/build/unity_src/materials_Unity.C...
Linking Library /home/sidharth/projects/def-jsong/sidharth/moose/modules/solid_mechanics/lib/libsolid_mechanics-opt.la...
Regex version mismatch, expected: 10.42 2022-12-11 actual: 10.32 2018-09-10
Regex version mismatch, expected: 10.42 2022-12-11 actual: 10.32 2018-09-10
Regex version mismatch, expected: 10.42 2022-12-11 actual: 10.32 2018-09-10
Regex version mismatch, expected: 10.42 2022-12-11 actual: 10.32 2018-09-10
Linking Executable /home/sidharth/projects/def-jsong/sidharth/moose/modules/solid_mechanics/solid_mechanics-opt...
/cvmfs/soft.computecanada.ca/gentoo/2023/x86-64-v3/usr/x86_64-pc-linux-gnu/binutils-bin/2.41/ld: /home/sidharth/projects/def-jsong/sidharth/moose/modules/solid_mechanics/lib/libsolid_mechanics-opt.so: undefined reference to `ScheiderBrittle::validParams()'
collect2: error: ld returned 1 exit status
make: *** [/home/sidharth/projects/def-jsong/sidharth/moose/framework/app.mk:457: /home/sidharth/projects/def-jsong/sidharth/moose/modules/solid_mechanics/solid_mechanics-opt] Error 1
[sidharth@beluga4 solid_mechanics]$ 
```

**Reply by GiudGiud** [reply]
This syntax has been deprecated for 3 years
```
template <>
InputParameters
validParams<ScheiderBrittle>()
```

please update the code

**Reply by sidharthsarmah** [reply]
@GiudGiud , please suggest in solid_mechanics:
```
*** ERROR ***
Object 'ADMatDiffusion' registered from multiple files: /Users/sidharthsarmah/MOOSE/projects/moose/modules/solid_mechanics/src/kernels/ADMatDiffusion.C and /Users/sidharthsarmah/MOOSE/projects/moose/framework/src/kernels/ADMatDiffusion.C
```

**Reply by GiudGiud** [reply]
> solid_mechanics/src/kernels/ADMatDiffusion.C

this should not be there. you must have added an `#include` with a .C file

**Reply by sidharthsarmah** [reply]
hi @GiudGiud , this is the .C included in solid_mech:
```
#include "ADMatDiffusion.h"

registerMooseObject("SolidMechanicsApp", ADMatDiffusion);

InputParameters
ADMatDiffusion::validParams()
{
  auto params = ADMatDiffusionBase<Real>::validParams();
  params.addClassDescription(
      "Diffusion equation kernel that takes an isotropic diffusivity from a material property");
  return params;
}

ADMatDiffusion::ADMatDiffusion(const InputParameters & parameters)
  : ADMatDiffusionBase<Real>(parameters)
{
}
```
.h:
```
//* This file is part of the MOOSE framework
//* https://www.mooseframework.org
//*
//* All rights reserved, see COPYRIGHT for full restrictions
//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
//*
//* Licensed under LGPL 2.1, please see LICENSE for details
//* https://www.gnu.org/licenses/lgpl-2.1.html

#pragma once

#include "ADMatDiffusionBase.h"

/**
 * Isotropic diffusion kernel that takes a diffusion coefficient of type
 * Real. All logic is implemnted in the ADMatDiffusionBase class
 * template.
 */
class ADMatDiffusion : public ADMatDiffusionBase<Real>
{
public:
  static InputParameters validParams();

  ADMatDiffusion(const InputParameters & parameters);
};
```

**Reply by GiudGiud** [reply]
you should remove this file it s not needed. We have MatDiffusion in the framework

**Reply by sidharthsarmah** [reply]
Thanks @GiudGiud
